### PR TITLE
[java] Add FormalParameterNamingConventions and LocalVariableNamingConventions

### PR DESCRIPTION
**Pull request by oowekyala**
These replace VariableNamingConventions in part. Still FieldNamingConventions to go. I'll add the release note and deprecate old rules when that last rule is added.

Refs #972

### CODE REVIEW

File: pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/codestyle/CodeStyleRulesTest.java [hunk 1]
```diff
@@ -37,14 +37,15 @@ public void setUp() {
...
         addRule(RULESET, "ForLoopShouldBeWhileLoop");
+        addRule(RULESET, "FormalParameterNamingConventions");
         addRule(RULESET, "GenericsNaming");
         addRule(RULESET, "IdenticalCatchBranches");
         addRule(RULESET, "IfElseStmtsMustUseBraces");
         addRule(RULESET, "IfStmtsMustUseBraces");
         addRule(RULESET, "LocalHomeNamingConvention");
         addRule(RULESET, "LocalInterfaceSessionNamingConvention");
         addRule(RULESET, "LocalVariableCouldBeFinal");
-        addRule(RULESET, "LongVariable");
```
Comment (jsotuyod) [inline pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/codestyle/CodeStyleRulesTest.java]: why is this removed?

Comment (oowekyala) [reply pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/codestyle/CodeStyleRulesTest.java]: Yeah this was a mistake
